### DISCUSSION TITLE
confounds.tsv in fmriprep vs halfpipe

### DISCUSSION THREAD

**Discussion by ryanwales** [HALFpipe/HALFpipe#448] [Q&A] [answered]
Hello,

Could you please explain the difference between the confounds.tsv in the fmriprep output folder versus the confounds.tsv in the halfpipe output directory? 

I'm curious why the values are slightly different for the same parameters between the two versions, for global signal, for example. Similarly, why do the motion_outliers columns appear to be all decimals instead of 0s and 1s? What transformation or calculation is done on these values? I believe grand mean scaling does not occur before the confounds.tsv is made, so the values should still be interpreted in their original units. 

Thank you for the clarification!

Best,
Ryan

**Comment by HippocampusGirl** [comment, marked as answer]
Hi @littlewhales17,

The confounds file output by fMRIPrep are the correct values to use if you are interested in the absolute values. The values output by HALFpipe are values that can be used for confound regression without accidentally re-introducing already removed artifacts. That means that grand mean scaling, ICA-AROMA and temporal filtering will all have been applied to the HALFpipe confounds file.

See also this section from our paper:

> ### 2.7 Confound time series removal
>
> fMRIPrep not only outputs a preprocessed image in standard space but also a spreadsheet with confound time series named confounds.tsv. These include the (derivatives of) motion parameters (squared), aCompCor components (Behzadi, Restom, Liau, & Liu, 2007), white matter signal, CSF signal, and global signal. A key consideration needs to be made when using fMRIPrep confound time series in conjunction with the preprocessing steps outlined in the previous section: Using confound time series as nuisance regressors with data that was temporally filtered or denoised can re-introduce the removed temporal or noise signals back into the voxel time series (Hallquist, Hwang, & Luna, 2013).
> 
> An example of this phenomenon may be regressing out a set of fMRIPrep confound time series after removing low-frequency drift via temporal filtering. In practice, this means setting up a regression model for each voxel, where the voxel time series is the dependent variable and the regressors are the confound time series. The regression will yield a weight for each confound time series, so that the total model explains the maximum amount of variance (under assumption of normality). After multiplying the confound time series with these weights, their products are summed to one time series containing the confound-related signal in that voxel. This time series is then subtracted from the original voxel time series to get the result (i.e., the regression residuals). However, if the confound time series happen to contain any low-frequency drift, then their weighted sum likely will as well. It follows that subtracting a time series with temporal drift from the temporally filtered voxel data will re-introduce temporal drift, independent of whether a temporal filter was applied before.
> 
> In HALFpipe, any (optional) denoising or filter applied to the voxel time series is also applied to the fMRIPrep confound time series. This way, previously removed variance is not re-introduced accidentally, because it has been removed from both sides of the regression equation. For example, when the user chooses to perform ICA-AROMA denoising, then the same denoising will be applied to the time fMRIPrep confound time series, and the same applies when using a temporal filter. Note that this means that the confound time series generated by HALFpipe will be different from the original fMRIPrep confound time series, and users should take care to use the appropriate file when running custom analyses.
